jenkins/bootstrap.py

```diff
@@ -818,9 +818,12 @@ def setup_root(call, root, repos, ssh, git_cache, clean):
                 os.path.join(root_dir, repo),
                 pull and pull or branch)
             checkout(call, repo, branch, pull, ssh, git_cache, clean)
+
     if len(repos) > 1:  # cd back into the primary repo
```

Comment:
So is this what you want?
```
len(repos) == 0 => cd root_dir
len(repos) == 1 => no cd
len(repos) > 1 => cd root_dir/repo1
```

Oh, we're relying on the fact that checkout() cds to the repo... How about dropping that implicit dep and doing something like:
```
cd root_dir
if repos:  cd repos.main
```

Reply:
I'll make the logic more clear